src/main/scala/beam/analysis/cartraveltime/CarRideStatsFromPathTraversalEventHandler.scala

```diff
@@ -122,14 +124,39 @@ class CarRideStatsFromPathTraversalEventHandler(
     buildStatistics(networkHelper, freeFlowTravelTimeCalc, iterationNumber, rideStats)
   }
 
+  private def createCarRideIterationGraph(
+    iterationNumber: Int,
+    rideStats: Seq[SingleRideStat],
+    mode: String
+  ): Unit = {
+
+    val hourAverageSpeed = rideStats.groupBy(stats => stats.departureTime.toInt / secondsInHour).map {
+      case (hour, statsList) => hour -> (statsList.map(_.speed).sum / statsList.size)
+    }
+
+    val maxHour = hourAverageSpeed.keys.max
+
+    val averageSpeed = (0 until maxHour).map(hourAverageSpeed.getOrElse(_, 0.0))
+
+    // generate the category dataset using the average travel times data
+    val dataSet = DatasetUtilities.createCategoryDataset("car", "", Array(averageSpeed.toArray))
+    createIterationGraphForAverageSpeed(dataSet, iterationNumber, mode)
+  }
+
   override def notifyIterationEnds(event: IterationEndsEvent): Unit = {
     val type2RideStats: Map[CarType, Seq[SingleRideStat]] = carType2PathTraversals.keys.map { carType =>
       carType -> calcRideStats(event.getIteration, carType)
     }.toMap
 
     type2RideStats.foreach {
       case (carType, stats) =>
-        writePersonalCarRideStats(event.getIteration, stats, carType)
+        writeCarRideStats(event.getIteration, stats, carType)
+    }
+
+    type2RideStats.get(CarType.RideHail) match {
```

Comment:
Why only for `CarType.RideHail` ? I think we have to create those graphs for all CarTypes

Reply:
now we are generating graphs for all car types